connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -190,4 +227,104 @@ public void commitRecord(SourceRecord record) {
             Checkpoint.unwrapGroup(record.sourcePartition()),
             System.currentTimeMillis() - record.timestamp());
     }
+
+    private void refreshIdleConsumerGroupOffset() {
+        Map<String, KafkaFuture<ConsumerGroupDescription>> consumerGroupsDesc = targetAdminClient
+            .describeConsumerGroups(consumerGroups).describedGroups();
+
+        for (String group : consumerGroups) {
+            try {
+                if (consumerGroupsDesc.get(group) == null) {
+                    // if consumerGroupsDesc does not contain this group, it should be the new consumer
+                    // group created at source cluster and its offsets should be sync-ed to target
+                    newConsumerGroup.add(group);
+                    continue;
+                }
+                ConsumerGroupDescription consumerGroupDesc = consumerGroupsDesc.get(group).get();
+                // sync offset to the target cluster only if the state of current consumer group is idle or dead
+                ConsumerGroupState consumerGroupState = consumerGroupDesc.state();
+                if (consumerGroupState.equals(ConsumerGroupState.EMPTY) || consumerGroupState.equals(ConsumerGroupState.DEAD)) {
+                    idleConsumerGroupsOffset.put(group, targetAdminClient.listConsumerGroupOffsets(group)
+                        .partitionsToOffsetAndMetadata().get().entrySet());
+                }
```

Comment:
I managed to get to work adding the DEAD consumer groups in the new consumer group list:

```suggestion
                if (consumerGroupState.equals(ConsumerGroupState.EMPTY)) {
                    idleConsumerGroupsOffset.put(group, targetAdminClient.listConsumerGroupOffsets(group)
                        .partitionsToOffsetAndMetadata().get().entrySet());
                } else if(consumerGroupState.equals(ConsumerGroupState.DEAD)){
                        newConsumerGroup.add(group);
                }
                
```

Reply:
thanks, adopted your suggestion. I will write unit tests to verify this scenario of adding new CG at source. If this is not a blocking factor, appreciated for more feedback

Reply:
Thanks @ning2008wisc; I'll let you know it I find any other corner cases in my tests.